Mage.Sets/src/mage/cards/i/IanMalcolmChaotician.java

```diff
@@ -0,0 +1,267 @@
+package mage.cards.i;
+
+import java.util.*;
+
+import mage.MageInt;
+import mage.MageObject;
+import mage.MageObjectReference;
+import mage.abilities.Ability;
+import mage.abilities.common.DrawNthCardTriggeredAbility;
+import mage.abilities.common.SimpleStaticAbility;
+import mage.abilities.effects.AsThoughEffectImpl;
+import mage.abilities.effects.AsThoughManaEffect;
+import mage.abilities.effects.OneShotEffect;
+import mage.cards.*;
+import mage.constants.*;
+import mage.game.CardState;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+import mage.players.ManaPoolItem;
+import mage.players.Player;
+import mage.util.CardUtil;
+import mage.watchers.Watcher;
+import mage.target.targetpointer.FixedTarget;
+
+/**
+ *
+ * @author jimga150
+ */
+public final class IanMalcolmChaotician extends CardImpl {
+
+    public IanMalcolmChaotician(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{U}{R}");
+        
+        this.supertype.add(SuperType.LEGENDARY);
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SCIENTIST);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(2);
+
+        // Whenever a player draws their second card each turn, that player exiles the top card of their library.
+        this.addAbility(new IanMalcolmChaoticianDrawTriggerAbility(), new IanMalcolmChaoticianWatcher());
+
+        // During each player's turn, that player may cast a spell from among the cards they don't own exiled with
+        // Ian Malcolm, Chaotician, and mana of any type can be spent to cast it.
+        Ability ability = new SimpleStaticAbility(new IanMalcolmChaoticianCastEffect());
+        ability.addEffect(new IanMalcolmChaoticianManaEffect());
+        this.addAbility(ability);
+    }
+
+    private IanMalcolmChaotician(final IanMalcolmChaotician card) {
+        super(card);
+    }
+
+    @Override
+    public IanMalcolmChaotician copy() {
+        return new IanMalcolmChaotician(this);
+    }
+}
+
+class IanMalcolmChaoticianDrawTriggerAbility extends DrawNthCardTriggeredAbility {
+    public IanMalcolmChaoticianDrawTriggerAbility() {
+        super(new IanMalcolmChaoticianExileEffect(), false, TargetController.ANY, 2);
+    }
+
+    private IanMalcolmChaoticianDrawTriggerAbility(final IanMalcolmChaoticianDrawTriggerAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        getEffects().setTargetPointer(new FixedTarget(event.getPlayerId()));
```

Comment:
the target pointer should only be set _after_ calling the super and confirming that trigger is returning true (well, it might not be a real bug here, but it's good style)